Integrations/McAfee_DXL/McAfee_DXL.py

```diff
@@ -0,0 +1,149 @@
+            text_file.write(params['broker_ca_bundle'])
+        with open(self.cert_file, "w") as text_file:
+            text_file.write(params['cert_file'])
+        with open(self.private_key, "w") as text_file:
+            text_file.write(params['private_key'])
+
+        if 'broker_urls' in params:
+            self.broker_urls = params['broker_urls'].split(',')
+        self.push_ip_topic = params.get('push_ip_topic')
+        self.push_url_topic = params.get('push_url_topic')
+        self.push_domain_topic = params.get('push_domain_topic')
+        self.push_hash_topic = params.get('push_hash_topic')
+        self.client = DxlClient(self.get_client_config())
+        self.client.connect()
+
+    def push_ip(self, ip, trust_level, topic):
+        if not is_ip_valid(ip):
+            raise ValueError(f'argument ip {ip} is not a valid IP')
+
+        trust_level_key = self.TRUST_LEVEL[trust_level]
+        if topic:
+            self.push_ip_topic = topic
+
+        self.send_event(self.push_ip_topic, f'ip:{ip};trust_level:{trust_level_key}')
+        return f'Successfully pushed ip {ip} with trust level {trust_level}'
+
+    def push_url(self, url, trust_level, topic):
+        trust_level_key = self.TRUST_LEVEL[trust_level]
+        if topic:
+            self.push_url_topic = topic
+
+        self.send_event(self.push_url_topic, f'url:{url};trust_level:{trust_level_key}')
+        return f'Successfully pushed url {url} with trust level {trust_level}'
+
+    def push_domain(self, domain, trust_level, topic):
+        trust_level_key = self.TRUST_LEVEL[trust_level]
+        if topic:
+            self.push_domain_topic = topic
+
+        self.send_event(self.push_domain_topic, f'domain:{domain};trust_level:{trust_level_key}')
+        return f'Successfully pushed domain {domain} with trust level {trust_level}'
+
+    def push_hash(self, hash_obj, trust_level, topic):
+        trust_level_key = self.TRUST_LEVEL[trust_level]
+        if topic:
+            self.push_ip_topic = topic
+
+        self.send_event(self.push_hash_topic, f'hash:{hash_obj};trust_level:{trust_level_key}')
+        return f'Successfully pushed hash {hash_obj} with trust level {trust_level}'
+
+    def get_client_config(self):
+        config = DxlClientConfig(
+            broker_ca_bundle=self.broker_ca_bundle,
+            cert_file=self.cert_file,
+            private_key=self.private_key,
+            brokers=[Broker.parse(url) for url in self.broker_urls]
+        )
+        config.connect_retries = CONNECT_RETRIES
+        config.reconnect_delay = RECONNECT_DELAY
+        config.reconnect_delay_max = RECONNECT_DELAY_MAX
+        return config
+
+    def send_event(self, topic, payload):
+        if not topic:
+            raise Exception(f'Error in {demisto.command()} topic field is required')
+
+        event = Event(topic)
+        event.payload = str(payload).encode()
+        self.client.send_event(event)
```

Comment:
on failure, will this return error or raise an exception?
should raise an exception.